gap/utils.gi

```diff
@@ -350,6 +350,64 @@ function(graph)
   fi;
 end);
 
+# Given a non-negative integer <n> and a homogeneous list <partition>,
+# this global function tests whether <partition> is a valid partition
+# of the list [1 .. n]. A valid partition of [1 .. n] is either:
+#
+# 1. A list of length <n> consisting of numbers in the range [0 .. n]
+# 2. A list of non-empty disjoint lists whose union is [1 .. n]
+#
+# If <partition> is a valid partition of [1 .. n] then this global function
+# returns the partition, in form 1 (converting it to this form if necessary).
+# If <partition> is invalid, then the function returns <fail>
+
+InstallGlobalFunction(DIGRAPHS_ValidateVertexPartition,
```

Comment:
I envisage `DIGRAPHS_ValidateVertexPartition` being used by the methods in `grahom.gi` as well, so I thought this would be the best place for it.